src/WorkflowManager/Storage/Services/DicomService.cs

```diff
@@ -237,9 +236,9 @@ public string GetValue(Dictionary<string, DicomValue> dict, string keyId)
             dict.TryGetValue(keyId, out var value);
 
             var result = string.Empty;
-            if (string.Equals(keyId, DicomTagConstants.PatientNameTag) || value.Vr.ToUpperInvariant() == "PN")
+            if (string.Equals(keyId, DicomTagConstants.PatientNameTag) || value?.Vr.ToUpperInvariant() == "PN")
             {
-                result = GetPatientName(value.Value);
+                result = GetPatientName(value!.Value);
```

Comment:
Instead of asserting it won't be null can we not convert the `dict.TryGetValue(keyId, out var value);` to be within an if statement and then anything that needs to access value go inside that?

```
var result = string.Empty;

if (dict.TryGetValue(keyId, out var value))
{
    // everything else here
}

return result;
```

Reply:
good spot even when with your comment took me a minute too see it :)